extensions/sql-database-projects/src/dialogs/createProjectFromDatabaseDialog.ts

```diff
@@ -0,0 +1,383 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as azdata from 'azdata';
+import * as vscode from 'vscode';
+import * as constants from '../common/constants';
+import * as newProjectTool from '../tools/newProjectTool';
+import * as mssql from '../../../mssql';
+
+import { IconPathHelper } from '../common/iconHelper';
+import { cssStyles } from '../common/uiConstants';
+import { ImportDataModel } from '../models/api/import';
+import { Deferred } from '../common/promise';
+
+export class CreateProjectFromDatabaseDialog {
+	public dialog: azdata.window.Dialog;
+	public importTab: azdata.window.DialogTab;
+	public sourceConnectionTextBox: azdata.InputBoxComponent | undefined;
+	private selectConnectionButton: azdata.ButtonComponent | undefined;
+	public sourceDatabaseDropDown: azdata.DropDownComponent | undefined;
+	public projectNameTextBox: azdata.InputBoxComponent | undefined;
+	public projectLocationTextBox: azdata.InputBoxComponent | undefined;
+	public folderStructureDropDown: azdata.DropDownComponent | undefined;
+	private formBuilder: azdata.FormBuilder | undefined;
+	private connectionId: string | undefined;
+	private toDispose: vscode.Disposable[] = [];
+	private initDialogComplete!: Deferred<void>;
+	private initDialogPromise: Promise<void> = new Promise<void>((resolve, reject) => this.initDialogComplete = { resolve, reject });
+
+	public createNewProjectCallBack: ((model: ImportDataModel) => any) | undefined;
+
+	constructor(private profile: azdata.IConnectionProfile | undefined) {
+		this.dialog = azdata.window.createModelViewDialog(constants.createProjectFromDatabaseDialogName);
+		this.importTab = azdata.window.createTab(constants.createProjectFromDatabaseDialogName);
+	}
+
+	public async openDialog(): Promise<void> {
+		this.initializeDialog();
+		this.dialog.okButton.label = constants.importDialogOkButtonText;
+		this.dialog.okButton.enabled = false;
```

Comment:
May as well just call `tryEnableImportButton()` at the end just to make sure it's consistent.

Reply:
In general, it's best to code defensively against bugs, which we can do here by keeping the logic as consistent and consolidated as possible to preempt potential future issues.  If someone does some refactoring here in the future that changes what the default state is (maybe by reordering how UI elements get registered or by having smart default values that make the "Okay" button enabled by default), they'll have to remember to change this as well.  Or we can preempt that by having this always calculate its value.

Ultimately, it's not a bug and more of a good habit but - I dunno - maybe I'm trying too hard to map how C# UIs are built to a different framework. :) Over there, you'd bind the enabled-ness to the UI validation itself so it always updates correctly for free and the dev can forget about it forever.

Reply:
That makes sense. Fixed it.

Reply:
That makes sense. Fixed it.